java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -0,0 +1,656 @@
+        Block thenBlock = currentBlock;
+        // process condition
+        currentBlock = createBranch(ifStatementTree, thenBlock, elseBlock);
+        buildCondition(ifStatementTree.condition(), thenBlock, elseBlock);
+        break;
+      }
+      case CONDITIONAL_EXPRESSION: {
+        ConditionalExpressionTree cond = (ConditionalExpressionTree) tree;
+        Block next = currentBlock;
+        // process else-branch
+        ExpressionTree elseStatement = cond.falseExpression();
+        currentBlock = createBlock(next);
+        build(elseStatement);
+        Block elseBlock = currentBlock;
+        // process then-branch
+        currentBlock = createBlock(next);
+        build(cond.trueExpression());
+        Block thenBlock = currentBlock;
+        // process condition
+        currentBlock = createBranch(cond, thenBlock, elseBlock);
+        buildCondition(cond.condition(), thenBlock, elseBlock);
+        break;
+      }
+      case VARIABLE:
+        currentBlock.elements.add(tree);
+        VariableTree variableTree = (VariableTree) tree;
+        if (variableTree.initializer() != null) {
+          build(variableTree.initializer());
+        }
+        break;
+      case MULTIPLY:
+      case DIVIDE:
+      case REMAINDER:
+      case PLUS:
+      case MINUS:
+      case LEFT_SHIFT:
+      case RIGHT_SHIFT:
+      case UNSIGNED_RIGHT_SHIFT:
+      case AND:
+      case XOR:
+      case OR:
+      case GREATER_THAN:
+      case GREATER_THAN_OR_EQUAL_TO:
+      case LESS_THAN:
+      case LESS_THAN_OR_EQUAL_TO:
+      case EQUAL_TO:
+      case NOT_EQUAL_TO:
+        BinaryExpressionTree binaryExpressionTree = (BinaryExpressionTree) tree;
+        currentBlock.elements.add(tree);
+        build(binaryExpressionTree.rightOperand());
+        build(binaryExpressionTree.leftOperand());
+        break;
+      case ASSIGNMENT:
+      case LEFT_SHIFT_ASSIGNMENT:
+      case RIGHT_SHIFT_ASSIGNMENT:
+      case AND_ASSIGNMENT:
+      case REMAINDER_ASSIGNMENT:
+      case UNSIGNED_RIGHT_SHIFT_ASSIGNMENT:
+      case OR_ASSIGNMENT:
+      case XOR_ASSIGNMENT:
+      case DIVIDE_ASSIGNMENT:
+      case MULTIPLY_ASSIGNMENT:
+      case PLUS_ASSIGNMENT:
+      case MINUS_ASSIGNMENT:
+        AssignmentExpressionTree assignmentExpressionTree = (AssignmentExpressionTree) tree;
+        currentBlock.elements.add(tree);
+        build(assignmentExpressionTree.variable());
+        build(assignmentExpressionTree.expression());
+        break;
+      case MEMBER_SELECT:
+        MemberSelectExpressionTree mse = (MemberSelectExpressionTree) tree;
+        currentBlock.elements.add(mse);
+        // int.class or String[].class are memberSelectExpression which expression part is not an expression.
+        if(!"class".equals(mse.identifier().name())) {
+          build(mse.expression());
+        }
+        break;
+      case CONDITIONAL_AND: {
+        BinaryExpressionTree e = (BinaryExpressionTree) tree;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 11. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)